diff --git a/component/mmdb/mmdb.go b/component/mmdb/mmdb.go
index 81644b00..b3ce0e5b 100644
--- a/component/mmdb/mmdb.go
+++ b/component/mmdb/mmdb.go
@@ -29,7 +29,8 @@ func LoadFromBytes(buffer []byte) {
 	ipOnce.Do(func() {
 		mmdb, err := maxminddb.FromBytes(buffer)
 		if err != nil {
-			log.Fatalln("Can't load mmdb: %s", err.Error())
+			log.Errorln("Can't load mmdb: %s", err.Error())
+			return
 		}
 		ipReader = IPReader{Reader: mmdb}
 		switch mmdb.Metadata.DatabaseType {
@@ -57,7 +58,8 @@ func IPInstance() IPReader {
 		log.Infoln("Load MMDB file: %s", mmdbPath)
 		mmdb, err := maxminddb.Open(mmdbPath)
 		if err != nil {
-			log.Fatalln("Can't load MMDB: %s", err.Error())
+			log.Errorln("Can't load MMDB: %s", err.Error())
+			return
 		}
 		ipReader = IPReader{Reader: mmdb}
 		switch mmdb.Metadata.DatabaseType {
@@ -79,7 +81,8 @@ func ASNInstance() ASNReader {
 		log.Infoln("Load ASN file: %s", ASNPath)
 		asn, err := maxminddb.Open(ASNPath)
 		if err != nil {
-			log.Fatalln("Can't load ASN: %s", err.Error())
+			log.Errorln("Can't load ASN: %s", err.Error())
+			return
 		}
 		asnReader = ASNReader{Reader: asn}
 	})
diff --git a/hub/executor/executor.go b/hub/executor/executor.go
index 5db0744a..f93857dc 100644
--- a/hub/executor/executor.go
+++ b/hub/executor/executor.go
@@ -481,7 +481,6 @@ func updateIPTables(cfg *config.Config) {
 	defer func() {
 		if err != nil {
 			log.Errorln("[IPTABLES] setting iptables failed: %s", err.Error())
-			os.Exit(2)
 		}
 	}()
 
